api/src/main/scala/Router.scala

```diff
@@ -0,0 +1,129 @@
+package org.wikiwatershed.mmw.geoprocessing
+
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
+import akka.http.scaladsl.model.HttpEntity
+import akka.http.scaladsl.server.Directives
+import akka.http.scaladsl.unmarshalling.Unmarshaller._
+import spray.json._
+import ch.megard.akka.http.cors.scaladsl.CorsDirectives._
+
+import scala.collection.JavaConversions._
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import com.typesafe.config.ConfigFactory
+
+class Router extends Directives {
```

Comment:
Instead of breaking out the router early, does it make sense to simplify things here so that we extend from `HttpApp` so that we get things like an `ActorSystem`, graceful termination, and a method for overriding routes for free?